platform/ios/src/MGLMapView.mm

```diff
@@ -3356,7 +3347,7 @@ - (void)_setCenterCoordinate:(CLLocationCoordinate2D)centerCoordinate edgePaddin
     }
     
     MGLMapCamera *camera = [self cameraForCameraOptions:cameraOptions];
-    if ([self.camera isEqualToMapCamera:camera])
+    if ([self.camera isEqualToMapCamera:camera] && UIEdgeInsetsEqualToEdgeInsets(_contentInset, insets))
```

Comment:
I don't think this is necessary ` && UIEdgeInsetsEqualToEdgeInsets(_contentInset, insets)` because you are already creating the `camera` with the insets. Check line 3324

Reply:
It is important to check both: this patch enables animation when padding only changes, too. When using the same camera, but setting new padding, view should animate.

What might be confusing you is that in [self cameraForCameraOptions:cameraOptions], value of cameraOptions.padding doesn't affect the camera. Whatever padding is specified, the camera remains the same as padding is not property of camera but property of the view.

CameraOptions padding is then used in easeTo, FlyTo... [L.3365](https://github.com/mapbox/mapbox-gl-native/pull/14664/files/3523033f00db4dd654dbf5df526082db28f2337d#diff-20502b6bc02986a4398d0da84b2d4b94R3365) and this makes that user of API: setCenterCoordinate gets the same result as earlier, when [self cameraForCameraOptions:cameraOptions] would alter the value of user specified cameraOptions.center based on the value of padding.

Reply:
Thank you for clarifying.